Model/Indexer/AsyncEventSubscriber.php

```diff
@@ -104,24 +99,34 @@ public function executeByDimensions(array $dimensions, Traversable $entityIds)
         $saveHandler = $this->indexerHandlerFactory->create(
             [
                 'data' => $this->data,
-                'scopeResolver' => $this->asyncEventScopeResolver
+                'scopeResolver' => $this->asyncEventScopeResolver,
+                'indexStructure' => $this->indexStructure
             ]
         );
 
-        $asyncEventIds = iterator_to_array($entityIds);
-
-        $this->batchSize = $this->deploymentConfig->get(
-            self::DEPLOYMENT_CONFIG_INDEXER_BATCHES . self::INDEXER_ID . '/partial_reindex'
-        ) ?? $this->batchSize;
-
-        $asyncEventBatches = array_chunk($asyncEventIds, $this->batchSize);
-
-        foreach ($asyncEventBatches as $asyncEventBatch) {
-            $this->processBatch(
-                $saveHandler,
+        if ($entityIds === null) {
+            $asyncEventDimension = $dimensions[0]->getValue();
```

Comment:
Is there any risk in `$dimensions = []` slipping through here?

Reply:
Not with the current implementation as it's not used anywhere. But it might be worth using a named property instead.

I'll update that. Good pickup

Reply:
Sorry I didn't elaborate, I meant is there any risk of `internal server error` happening from the array indexing issue if `!isset($dimensions[0])` or now `!isset($dimension['ae'])`